includes/Grrr/SimplyStaticDeploy/Config/RequiredFields.php

```diff
@@ -0,0 +1,36 @@
+<?php namespace Grrr\SimplyStaticDeploy\Config;
+
+use Garp\Functional as f;
+
+final class RequiredFields {
+
+    const REQUIRED_FIELDS = [
+        'aws' => [
+            'key',
+            'secret',
+            'region',
+            'bucket',
+        ],
+        'url',
+    ];
+
+    public static function getMissingOptions(array $data, array $requiredFields = self::REQUIRED_FIELDS, string $prefix = ''): array {
```

Comment:
```suggestion
    public static function get_missing_options(array $data, array $requiredFields = self::REQUIRED_FIELDS, string $prefix = ''): array {
```

Took me a couple of minutes to fully read it, but looks like what we need 👍

PS: in theory it might also have a recursive child function which is called, making the `getMissingOptions` parameters simpler. But only a thought, not something to spend time on I'd say.

Reply:
Yeah, I know, the function signature is not really clear.
The problem with a closure is calling it recursively ([I blogged about this once 😉](https://grrr.tech/posts/php-recursion/)). 
I could do it with the Y combinator from Garp\Functional but that's also quite a bit of code to mentally keep track of. 

Anyways. If I think of something and care to spend some time I'll pick it up again.

Reply:
Ah, yeah. I get where you're heading it 😒